lib/hash_pipe/array_conversion.rb

```diff
@@ -0,0 +1,27 @@
+Array.class_eval do
+  # Creates a deep object out of an array
+  #
+  # hash = ["errors", "something", "somethingElse", "password",  ["VALIDATION_STATE_PASSWORD"]]
+  #
+  # Could be turned into an object usign this
+  #
+  # hash.nest
+  #
+  # And the end result would look like this:
+  #
+  # "errors" => {
+  #   "something" => {
+  #     "somethingElse" => {
+  #       "password" => ["VALIDATION_STATE_PASSWORD"]
+  #     }
+  #   }
+  # }
+  #
+  def nest
+    self.each do |elm|
+      raise ArgumentError, "Name too big" if elm.nil?
+    end
```

Comment:
Not sure I understand the exception message here. Why "too big"?

This could also be cleaned up a little using `.any?`:

```ruby
raise "..." if any? &:nil?
```